test/libdnf/base_test_case.cpp

```diff
@@ -85,6 +109,24 @@ libdnf::rpm::Package BaseTestCase::get_pkg(const std::string & nevra, const char
 }
 
 
+libdnf::rpm::Package BaseTestCase::get_pkg_i(const std::string & nevra, size_t index) {
+    libdnf::rpm::PackageQuery query(base);
+    query.filter_nevra({nevra});
+
+    if (query.size() <= index) {
+        CPPUNIT_FAIL(
+            sformat("Package index {} out of bounds for \"{}\", query packages:{}", index, nevra, to_string(query)));
+    }
+
+    auto it = query.begin();
+    while (index-- > 0) {
+        it++;
```

Comment:
Note: In general, it is better to use prefix operator instead of the postfix one when the return value is not used. The postfix operator is usually implemented as a prefix operator plus additional work - storing the original value before incrementing and than returning the original value.
Because the returned value is not used the compiler can optimize - eliminate extra code, but sometime it needs LTO or it is not done at all.